patches/server/0949-Implement-CraftScheduler-getAsyncExecutor-API.patch

```diff
@@ -0,0 +1,299 @@
++
++        // Wait for plugins to close their threads
+         int pollCount = 0;
+ 
+-        // Wait for at most 5 seconds for plugins to close their threads
+-        while (pollCount < 10*5 && getScheduler().getActiveWorkers().size() > 0) {
++        // During reload, wait for at most 2.5 seconds, in shutdown, 5 seconds
++        long sleepTime = (reload) ? 50 : 100;
++
++        while (pollCount < 50 && !getScheduler().isAllFinished()) {
+             try {
+-                Thread.sleep(100);
++                Thread.sleep(sleepTime);
+             } catch (InterruptedException e) {}
+             pollCount++;
+         }
+ 
+-        List<BukkitWorker> overdueWorkers = getScheduler().getActiveWorkers();
++        List<BukkitWorker> overdueWorkers = this.getScheduler().getActiveWorkers();
+         for (BukkitWorker worker : overdueWorkers) {
+             Plugin plugin = worker.getOwner();
+-            String author = "<NoAuthorGiven>";
+-            if (plugin.getDescription().getAuthors().size() > 0) {
+-                author = plugin.getDescription().getAuthors().get(0);
++            warnNotShuttingDownAsyncTasks(plugin, reload);
++            if (reload && console.isDebugging()) {
++                io.papermc.paper.util.TraceUtil.dumpTraceForThread(worker.getThread(), "still running");
+             }
+-            getLogger().log(Level.SEVERE, String.format(
+-                "Nag author: '%s' of '%s' about the following: %s",
+-                author,
+-                plugin.getDescription().getName(),
+-                "This plugin is not properly shutting down its async tasks when it is being shut down. This task may throw errors during the final shutdown logs and might not complete before process dies."
+-            ));
+         }
++        for (Plugin plugin : getScheduler().getUnfinishedFromExecutors()) {
++            warnNotShuttingDownAsyncTasks(plugin, reload);
++        }
++    }
++
++    private void warnNotShuttingDownAsyncTasks(Plugin plugin, boolean reload) {
++        String message;
++        if (reload) {
++            message = "This plugin is not properly shutting down its async tasks when it is being reloaded. "
++                    + "This may cause conflicts with the newly loaded version of the plugin";
++        } else {
++            message = "This plugin is not properly shutting down its async tasks when it is being shut down. "
++                    + "This task may throw errors during the final shutdown logs and might not complete before process dies.";
++        }
++        getLogger().log(Level.SEVERE, String.format(
++            "Nag author(s): '%s' of '%s' about the following: %s",
++            plugin.getDescription().getAuthors(),
++            plugin.getDescription().getFullName(),
++            message
++        ));
++    }
++
++    public void waitForAsyncTasksShutdown() {
++        waitForAsyncTasksToShutdown(false);
+     }
+     // Paper end
```

Comment:
Looks like there's a bit of conflict with a pre-existing paper patch. Can you please merge some of these changes into the already existing ``Wait for Async Tasks during shutdown`` patch?
This should hopefully pretty up the diff here.